source/gui/addonGui.py

```diff
@@ -378,7 +380,8 @@ def onClose(self,evt):
 		for addon in self.curAddons:
 			if (addon.isPendingInstall or addon.isPendingRemove
 				or addon.isDisabled and addon.isPendingEnable
-				or addon.isRunning and addon.isPendingDisable):
+				or addon.isRunning and addon.isPendingDisable
+				or (not addon.isRunning and addon.isPendingDisable and globalVars.appArgs.disableAddons)):

```

Comment:
Why this change?

Reply:
Because if an add-on was enabled, and you decide to disable it (while add-ons are globally disabled), we want to show the restart dialog. Sorry again, this looks confusing, and I think it is at some point.